akka-persistence/src/main/scala/akka/persistence/Persistence.scala

```diff
@@ -344,6 +344,14 @@ class Persistence(val system: ExtendedActorSystem) extends Extension {
       case None       => throw new IllegalArgumentException(s"Unknown plugin actor $journalPluginActor")
     }
 
+  private[akka] final def extensionIdFor(journalPluginActor: ActorRef): String =
+    pluginExtensionId.get().collectFirst {
+      case (id, ext) if ext(system).actor == journalPluginActor => id
+    } match {
+      case Some(conf) => conf
+      case None       => throw new IllegalArgumentException(s"Unknown plugin actor $journalPluginActor")
+    }
```

Comment:
`getOrElse` would be more straightforward as we are not doing anything in the `Some` case, just unwrapping it.

Reply:
Yeah, I agree, not hot path in any way, mostly followed how the other similar methods are done. Should add internal api markers though.